08-components-lab/src/02-numberguessing.jsx

```diff
@@ -1,17 +1,46 @@
-/**
- * Lab 02:
- *
- * Implement a number guessing game component.
- *
- * The component should provide an input field for the user to guess a
- * number picked randomly by the computer (from 1 to 1,000).
- * When typing a number, the component should print "too low" or "too high" 
- * as a hint to the user.
- *
- * To make the game interesting, computer should sometimes cheat and print wrong messages.
- */
-
-import React from 'react';
-
-export default React.createClass({
-});
+import React from "react";
+
+export default class NumberGuessing extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            userGuess: 0,
+            randomNumber: Math.floor((Math.random() * 1000) + 1),
+            validateMsg: "",
+        };
+
+        console.log(this.state.randomNumber);
+    }
+
+    newNumberGuess(event) {
+        var userGuess = Number(event.target.value);
+        this.setState({
+            userGuess: userGuess,
+        });
+
+        this.validateUserGuess(userGuess);
+    }
+
+    validateUserGuess(userGuess) {
+        var validateMsg;
+        if (userGuess > this.state.randomNumber || userGuess % 7 == 0) { // greater
+            validateMsg = "greater"
+        } else if (userGuess < this.state.randomNumber || userGuess % 5 == 0) { // less
+            validateMsg = "smaller"
+        } else { // equal
+            validateMsg = "equal!"
+        }
+
+        this.setState({validateMsg});
+    }
+
+    render() {
+        return (<div>
+            <input onChange={(e) => this.newNumberGuess(e)}/>
+            <br/>
```

Comment:
I advice not to use <br/> tags. If you want proper separation between html elements better do it with css or use inline style with "style" prop.